### ISSUE TITLE
How consul runs on deploy ?

### ISSUE THREAD

**Issue by GuillaumeDievart** [hashicorp/otto#269]
Hi,

I test to deploy a simple PHP app, with a dependency (redis docker), but once that my apps deployed, I don't see the redis app on consul, I missed anything ?

My Appfile(s):
https://gist.github.com/GuillaumeDievart/b334f93c2053870090c4

Sorry for my poor english ...

Also, when I try to build my redis app, nothing is done, and the build status stays "NOT BUILT"

**Comment by adambarthelson** [comment]
I'm not sure if you need "Project" and "Infrastructure" in your Appfile for your Redis container, someone correct me if I'm wrong.  Were you able to get this working locally with ```otto dev```?

If you ```otto dev ssh```, and then run ```dig +short redis.service.consul.``` is an address returned?

If not, what version of Otto do you have?  I had a similar issue with Consul on v0.1.1, but found it was resolved in v0.1.2.

**Comment by GuillaumeDievart** [comment]
Hi @adambarthelson ,

this configuration works on dev:
```
vagrant@precise64:/vagrant$ dig +short redis.service.consul
10.0.2.15
```

The problem occurs because I used a simple flavor instead vpc ?

I use the v0.1.1, I'll tried with v0.1.2.

Thanks !

**Comment by GuillaumeDievart** [comment]
Same issue with this simple Appfile, my dependency is not reachable:
```
application {
    name = "test"
    type = "php"

    dependency { source = "github.com/hashicorp/otto/examples/mongodb" }
}

project {
    name = "test"
    infrastructure = "test"
}

infrastructure "test" {
  type = "aws"
  flavor = "simple"
}
```
What I missed ?

**Comment by GuillaumeDievart** [comment]
Nobody for help me ? ping @mitchellh

**Comment by adambarthelson** [comment]
There's a known issue with the MongoDB example dependency in the repository.

If you to set up a MongoDB service, I've had success creating a folder to my project's directory for dependencies.

For example, in ```/yourProject/mongodb/```, add an Appfile:
```
application {
    name = "mongodb"
    type = "docker-external"
}

customization "docker" {
    image = "mongo:latest"
    run_args = "-p 27017:27017"
}
```

Then in ```/yourProject/Appfile``` you can do something like this:
```
application {
  name = "test"
  type = "php"

  dependency { source = "mongodb" }
}
```

**Comment by tonysm** [comment]
Same issue here. I tried to launch a PHP application with a redis docker dependency.. works locally, fails on AWS. I noticed that the Packer build doesn't mention the redis docker dependency, which the `otto dev` command mentions during the vagrant machine provisioning.

I tried doing the way it's mentioned in the getting started docs, but no success. Also tried how @adambarthelson said here, works fine locally, but can't resolve the service on AWS either.

Also using the `simple` flavor here, going to try with the `vpc` one.

Here is the repo: https://github.com/tonysm/otto-laravel
